### PR TITLE
major refactor

### PR DESCRIPTION
**Pull request by daroga0002**
This is major refactor of modules. It changes:
- adding additional `tags` variable to allow tagging by customer common tags 
- remove non required quotations
- standarize tags and name_prefix
- add descriptions to inputs and outputs
- enforce types across variables

### CODE REVIEW

**Comment by daroga0002** [comment]
@zhiqiangjin  could you review this

File: aviatrix-controller-initialize/variables.tf [hunk 1]
```diff
@@ -1,32 +1,73 @@
...
+  description = "Customer license ID"
+  default     = ""
+}
+
+variable controller_launch_wait_time {
+  description = "Controller launch wait time"
+  default     = 210
+}
+
+variable vpc {
```
Comment (arvindsree) [inline aviatrix-controller-initialize/variables.tf:57]: vpc_id

File: aviatrix-controller-initialize/variables.tf [hunk 2]
```diff
@@ -1,32 +1,73 @@
...
+  type        = string
+  description = "VPC in which you want launch Aviatrix controller"
+}
+
+variable subnet {
```
Comment (arvindsree) [inline aviatrix-controller-initialize/variables.tf:62]: subnet_id

File: aviatrix-controller-initialize/variables.tf [hunk 3]
```diff
@@ -1,32 +1,73 @@
...
+  type        = string
+  description = "Subnet in which you want launch Aviatrix controller"
+}
+
+locals {
+  common_tags = merge(
+    var.tags, {
+      module    = "aviatrix-controller-build"
```
Comment (arvindsree) [inline aviatrix-controller-initialize/variables.tf:70]: aviatrix-controller-initialize

File: aviatrix-controller-iam-roles/variables.tf [hunk 4]
```diff
@@ -39,8 +52,16 @@ EOF
     ]
 }
 EOF
+
+  common_tags = merge(
+    var.tags, {
+      module    = "aviatrix-controller-build"
```
Comment (arvindsree) [inline aviatrix-controller-iam-roles/variables.tf:58]: aviatrix-controller-iam-roles

File: aviatrix-controller-build/variables.tf [hunk 5]
```diff
@@ -1,60 +1,99 @@
...
+      module    = "aviatrix-controller-build"
+      Createdby = "Terraform+Aviatrix"
+  })
 }
 
-data "http" "avx_iam_id" {
-  url             = "https://s3-us-west-2.amazonaws.com/aviatrix-download/AMI_ID/ami_id.json"
+
+data http avx_iam_id {
+  url = "https://s3-us-west-2.amazonaws.com/aviatrix-download/AMI_ID/ami_id.json"
```
Comment (arvindsree) [inline aviatrix-controller-build/variables.tf:96]: Use consistent indexing. Otherwise don't change

File: aviatrix-controller-initialize/main.tf [hunk 6]
```diff
@@ -22,7 +18,7 @@ resource "aws_iam_role" "iam_for_lambda" {
 EOF
 }
 
-resource "aws_iam_policy" "lambda-policy" {
+resource aws_iam_policy lambda-policy {
```
Comment (arvindsree) [inline aviatrix-controller-initialize/main.tf:21]: Do all versions of terraform 0.12.x support this?

File: aviatrix-controller-iam-roles/variables.tf (hunk 4 above)
Comment (daroga0002) [reply aviatrix-controller-iam-roles/variables.tf]: good catch, changed

File: aviatrix-controller-initialize/variables.tf (hunk 3 above)
Comment (daroga0002) [reply aviatrix-controller-initialize/variables.tf]: good catch, changed

File: aviatrix-controller-initialize/variables.tf (hunk 2 above)
Comment (daroga0002) [reply aviatrix-controller-initialize/variables.tf]: good catch, changed

File: aviatrix-controller-initialize/variables.tf (hunk 1 above)
Comment (daroga0002) [reply aviatrix-controller-initialize/variables.tf]: good catch, changed

File: aviatrix-controller-build/variables.tf (hunk 5 above)
Comment (daroga0002) [reply aviatrix-controller-build/variables.tf:96]: Looks this is how `terraform fmt` is  doing it, I have done test by manually adjusting this and then running `terraform fmt` which modified this as it is here. 

Should I rely on `terraform fmt` or I should manually make it consistent?

File: aviatrix-controller-initialize/main.tf (hunk 6 above)
Comment (daroga0002) [reply aviatrix-controller-initialize/main.tf:21]: I think so as I tested this also on 0.12.0

File: aviatrix-controller-build/variables.tf (hunk 5 above)
Comment (arvindsree) [reply aviatrix-controller-build/variables.tf:96]: I see. That should be fine then

**Comment by arvindsree** [comment]
Thanks for the PR and changes

**Pull request merged**